src/wallet/scriptpubkeyman.cpp

```diff
@@ -281,9 +281,10 @@ void LegacyScriptPubKeyMan::KeepDestination(int64_t index)
     KeepKey(index);
```

Comment:
In commit "Key pool: Change ReturnDestination interface to take address instead of key" (06d411488c1c669f00a1c947416267941dbb7f7b)

Why doesn't this need to call m_reserved_key_to_index.erase?

Reply:
It should.

This wasn't necessarily a bug as it doesn't effect behavior.